### PR TITLE
Split system label functionality into SystemLabel and SystemLabelCount

### PR DESCRIPTION
**Pull request by rowanseymour**
Currently `SystemLabel` is both treated as a virtual label (i.e. when we create a new org, we create a single instance for the Inbox, etc), and as a squashable count. This splits it into `SystemLabel` as a purely virtual label, and `SystemLabelCount` as the squashable count model.

I decided against merging `Label` and `SystemLabel` for now as this would complicate the db triggers. Curiously `Label` still uses count tracking via updates but I don't think we've seen deadlock issues because of this. Probably because it's unlikely an incoming message would be updated simultaneously by more than one thread.

### CODE REVIEW

**Comment by nicpottier** [comment]
Oh man, good catch on us using dump counts on SystemLabels, seems we should update that or we'll be totally slammed as soon as anybody adds a label msg action in a flow.

**Comment by rowanseymour** [comment]
@nicpottier you mean changing `Label` to use squashable counts? I think it'll be cleanest to introduce a new model like `LabelCount` since `SystemLabel` and `Label` are still different things with different mechanisms for tracking.

**Comment by nicpottier** [comment]
Yep yep, totally fine with a LabelCount

File: temba/api/tests/test_v2.py [hunk 1]
```diff
@@ -1678,7 +1678,7 @@ def test_labels(self):
         important.toggle_label([msg], add=True)
 
         # no filtering
-        with self.assertNumQueries(NUM_BASE_REQUEST_QUERIES + 1):
+        with self.assertNumQueries(NUM_BASE_REQUEST_QUERIES + 2):
```
Comment (rowanseymour) [inline temba/api/tests/test_v2.py]: Extra db hit to get label counts from separate model

**Comment by rowanseymour** [comment]
@nicpottier please re-review

File: temba/msgs/migrations/0086_label_counts.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  SELECT label_id, _count, FALSE FROM msgs_msg_labels WHERE msgs_msg_labels.msg_id = _msg_id;
+END;
+$$ LANGUAGE plpgsql;
+
+----------------------------------------------------------------------
+-- Reset (i.e. zero-ize) system label counts of the given type across all orgs
+----------------------------------------------------------------------
+CREATE OR REPLACE FUNCTION temba_reset_system_labels(_label_types CHAR(1)[]) RETURNS VOID AS $$
+BEGIN
+  DELETE FROM msgs_systemlabelcount WHERE label_type = ANY(_label_types);
```
Comment (nicpottier) [inline temba/msgs/migrations/0086_label_counts.py]: Was going to comment that maybe TRUNCATE is faster here but:
 1) this is likely never going to be called anyways and
 2) seems there are some scenarios where that isn't true.

So godspeed. :)

File: temba/msgs/models.py [hunk 3]
```diff
@@ -1696,41 +1669,44 @@ def get_queryset(cls, org, label_type, exclude_test_contacts=True):
...
+        sql = """
+        WITH deleted as (
+            DELETE FROM %(table)s WHERE "org_id" = %%s AND "label_type" = %%s RETURNING "count"
+        )
+        INSERT INTO %(table)s("org_id", "label_type", "count", "is_squashed")
+        VALUES (%%s, %%s, GREATEST(0, (SELECT SUM("count") FROM deleted)), TRUE);
+        """ % {'table': cls._meta.db_table}
 
-        return counts_by_type
+        return sql, (distinct_set.org_id, distinct_set.label_type) * 2
```
Comment (nicpottier) [inline temba/msgs/models.py]: What is this doing? Not familiar with this syntax.

File: temba/msgs/migrations/0086_label_counts.py (hunk 2 above)
Comment (nicpottier) [inline temba/msgs/migrations/0086_label_counts.py]: Oh and I see why there's a select in there, so NM.

File: temba/msgs/models.py (hunk 3 above)
Comment (rowanseymour) [inline temba/msgs/models.py]: `[a, b] * 3 == [a, b, a, b, a, b]`

File: temba/msgs/migrations/0086_label_counts.py (hunk 2 above)
Comment (rowanseymour) [inline temba/msgs/migrations/0086_label_counts.py]: Yeah we're only nuking counts for items from tables which are being truncated

File: temba/msgs/models.py (hunk 3 above)
Comment (nicpottier) [inline temba/msgs/models.py]: Thought it might be that, wtf python.

**Pull request merged**